Dockerfile

```
@@ -1,58 +1,61 @@
 # Setup dev environment
-FROM ubuntu:18.04 AS environment
-
+FROM node:16-bullseye as build-stage
 ENV DEBIAN_FRONTEND noninteractive
+RUN apt-get update && apt-get install -y git libxkbfile-dev libsecret-1-dev bash libsecret-1-0  maven openjdk-11-jdk
 
-RUN apt-get update && \
-	apt-get upgrade -y && \
-	apt-get install -y default-jdk maven && \
-	apt-get install wget build-essential cmake libopenblas-dev gnupg curl make git g++-multilib clangd-10 gdb libsecret-1-dev -y
-
-RUN update-alternatives --install /usr/bin/clangd clangd /usr/bin/clangd-10 100
-
-RUN curl -fsSL https://deb.nodesource.com/setup_16.x | bash - && \
-	apt-get install nodejs -y && \
-	npm install -g yarn
+# Theia build dependencies
+RUN  export DEBIAN_FRONTEND=noninteractive \
+	&& apt-get -y install --no-install-recommends \
+	software-properties-common \
+	libxkbfile-dev \
+	libsecret-1-dev \
+	build-essential libssl-dev
 
-# Build the Java backend
-FROM environment AS backend
 
 WORKDIR /coffee-editor
 
 COPY ./backend ./backend
+COPY ./client ./client
 
 WORKDIR /coffee-editor/backend/releng/org.eclipse.emfcloud.coffee.parent/
-
 RUN mvn clean verify
 
+WORKDIR /coffee-editor/client
 
-# Build frontend
-FROM environment AS frontend
+RUN yarn install &&\
+	yarn copy:servers &
 
-WORKDIR /coffee-editor
+FROM node:16-bullseye-slim as production-stage
 
-COPY ./client ./client
 
-WORKDIR /coffee-editor/client
+# Theia dependencies/Java
+RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+	&& apt-get -y install --no-install-recommends \
+	software-properties-common \
+	libxkbfile-dev \
+	libsecret-1-dev openjdk-11-jdk \
+	build-essential libssl-dev  wget gnupg git mvn
```

```suggestion
	build-essential libssl-dev  wget gnupg git
```
Installing `mvn` fails for me and is not part of the k8s dockerfile. Can it be removed?

---

Sorry, should be `maven`. I will update it

---

But do we actually need it in the production stage? Just java to run the jars should be enough, right? The k8s dockerfile seems to also be able to run without it.

---

Yes or running its fine but you cannot build without it. I think the "normal" image should also contain all dev dependencies so that you could theoretically develop the coffee editor in a container. Otherwise there is no point in providing to separate images. 
But if we don't need that we can also reduce this to one dockerfile that is optimized for the cluster

---

No thats fine. Thanks for the clarification :+1: